draft-ietf-quic-tls.md

```diff
@@ -933,32 +931,49 @@ to QUIC.
 ## Packet Number Protection {#pn-encrypt}
 
 QUIC packets are protected using a key that is derived from the current set of
-secrets.  The key derived using the "pn" label is used to protect the packet
-number from casual observation.  The packet number protection algorithm depends
-on the negotiated AEAD.
+secrets.  A packet protection key is used to protect the packet number from
+casual observation.  The packet number protection key is derived from the first
+packet protection secret using the label "pn".  This key is produced using the
+same process as the AEAD key and IV, but the value is not updated with each key
+update ({{key-update}}).
+
+~~~
+pn_key = QHKDF-Expand(pp_secret<0>, "pn", pn_key_length)
+~~~
+
+That is, for the algorithms defined in this document, the packet protection key
+is determined to be:
+
+~~~
+client_pn_key  = QHKDF-Expand(client_pp_secret<0>, "pn", 16)
+server_pn_key  = QHKDF-Expand(server_pp_secret<0>, "pn", 16)
```

Comment:
My understanding is that `pp_secret` only exists for 1-RTT. Do we need to define similar constructs for handshake secret and 0-RTT secret?

Reply:
These are examples.  The same secret that you would use for getting a key or IV should work for 0-RTT and the handshake.  That probably needs work though.

Reply:
Thank you for the clarification. I think my confusion came from the statement saying
> The packet number protection key is derived from **the first packet protection secret** using the label "pn".

which seem to imply a 1-RTT secret, since section 5.3.4 (1-RTT secrets) has the following sentence:
> **The initial client packet protection secret** is exported from TLS using the exporter label “EXPORTER-QUIC client 1rtt”.